scripts/bmp.py

```diff
@@ -0,0 +1,62 @@
+"""
+Python classes for BMP 
+"""
+
+from enum import Enum
+import ipaddress
+import json
+
+class MessageType(Enum):
+    ROUTE_MONITORING = 0
+    STATISTICS = 1
+    PEER_DOWN = 2
+    PEER_UP = 3
+    INITIATION = 4
+    TERMINATION = 5
+    ROUTE_MIRRORING = 6
+
+def parse_message(msg_json):
+    msg_dict = json.loads(msg_json)
+    header = msg_dict['Header']
+    msg_type = MessageType(header['Type'])
+    if msg_type == MessageType.PEER_DOWN:
+        return PeerDownMessage(msg_dict)
+    elif msg_type == MessageType.PEER_UP:
+        return PeerUpMessage(msg_dict)
+    else:
+        return Message(msg_type, msg_dict)
+
+class Message:
+    def __init__(self, msg_type, msg_dict):
+        self._type = msg_type
+        src_id = msg_dict['PeerHeader']['PeerBGPID']
+        self._src_id = (None if src_id == '' else ipaddress.ip_address(src_id))
+        src_as = msg_dict['PeerHeader']['PeerAS']
+        self._src_as = (None if src_as == 0 else src_as)
```

Comment:
I should have been clearer. My question was whether `src_id` is empty iff src_as is zero? I guess the JSON parser has no knowledge whether a field is a number or a string, so I was curious about the values, rather than the presence of fields.

Reply:
Yes, `src_id` will always be the empty string when `src_as` is zero. Both are required in a proper BGP configuration, so you can't have one but not the other.